packages/api/amplify_api_dart/lib/src/api_plugin_impl.dart

```diff
@@ -158,10 +182,90 @@ class AmplifyAPIDart extends APIPluginInterface with AWSDebuggable {
     );
   }
 
+  EndpointConfig _getEndpointConfig(EndpointType type, String? apiName) {
+    if (type == EndpointType.graphQL) {
+      if (_dataConfig == null) {
+        throw ConfigurationError(
+          'No GraphQL API endpoint found.',
+        );
+      }
+      DataOutputs config;
+      if (apiName != null) {
+        config = _dataConfig.entries
+            .firstWhere(
+              (config) => config.key == apiName,
+              orElse: () => throw ConfigurationError(
+                'No GraphQL API endpoint found matching apiName $apiName.',
+              ),
+            )
+            .value;
+      } else {
+        if (_dataConfig.length > 1) {
+          throw ConfigurationError(
+            'Multiple GraphQL API endpoints defined. Pass apiName parameter to specify '
+            'which one to use.',
+          );
+        }
+        config = _dataConfig.values.first;
+        apiName = _dataConfig.keys.first;
+      }
+      return EndpointConfig(
+        apiName,
+        AWSApiConfig(
```

Comment:
`AWSApiConfig` is part of `AmplifyConfig` (`AWSApiPluginConfig` extends `ConfigMap<AWSApiConfig>`, `AWSApiPluginConfig` is part of `ApiConfig`).

Can we do this without mapping back to AmplifyConfig types? We are duplicating mapping logic. We already have logic to map from AmplifyConfig to AmplifyOutputs. This is adding logic to do the inverse.

Reply:
I suggest making a common type that DataOutputs and RestApiOutputs extend and mapping to that instead.